pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -375,6 +375,11 @@ func getLatestConfig(url string, iteration int,
 						ts.Format(time.RFC3339Nano))
 					getconfigCtx.readSavedConfig = true
 					getconfigCtx.configGetStatus = types.ConfigGetReadSaved
+					// When getting the config successfully for the first time (read from file in this case), consider
+					// the device as ready to start apps. Hence, count the app delay timeout from now.
+					if getconfigCtx.zedagentCtx.delayBaseTime.IsZero() {
+						getconfigCtx.zedagentCtx.delayBaseTime = time.Now()
+					}
```

Comment:
To me it is a lot simpler if this logic lives in zedmanager and there are no changes what so ever in zedagent/*.go
That means zedmanager can set its local notion of delayBaseTime in handleZedAgentStatusImpl()

Reply:
I can try to move it to zedmanager, ok.

Reply:
Moved into Zedmanager.